Source/JavaScriptCore/bytecode/ArithProfile.cpp

```diff
@@ -76,10 +76,17 @@ bool ArithProfile<BitfieldType>::shouldEmitSetDouble() const
 }
 
 template<typename BitfieldType>
-void ArithProfile<BitfieldType>::emitSetDouble(CCallHelpers& jit) const
+void ArithProfile<BitfieldType>::emitSetDouble(CCallHelpers& jit, GPRReg scratchGPR) const
 {
-    if (shouldEmitSetDouble())
+    if (shouldEmitSetDouble()) {
+#if CPU(ARM64)
+        jit.move(CCallHelpers::TrustedImm32(ObservedResults::Int32Overflow | ObservedResults::Int52Overflow | ObservedResults::NegZeroDouble | ObservedResults::NonNegZeroDouble), scratchGPR);
```

Comment:
Just curious: Was the issue here that this value isn't a valid logic immediate?

Reply:
This is,

> During working on this, we also found that ArithProfile::emitSetDouble is materializing a pointer twice
for ARM64. This patch fixes it.

This part.